extensions/wikia/Recirculation/js/utils.js

```diff
@@ -44,8 +44,10 @@ define('ext.wikia.recirculation.utils', [
 	}
 
 	function buildLabel(element, label) {
-		var slot = $(element).parent().index() + 1;
-		
+		// This is a work around to make sure we only get the index based on actual list items
+		var $listItem = $(element).parent();
+		var slot = $listItem.closest('ul').children('li').index($listItem) + 1;
```

Comment:
I think it would be more straight-forward to assign the numbers in the place we build the list and use data params for it.